kornia-py/src/apriltag.rs

```diff
@@ -0,0 +1,449 @@
+    }
+
+    #[setter]
+    pub fn set_normal_border(&mut self, value: bool) {
+        self.0.normal_border = value;
+    }
+
+    #[getter]
+    pub fn get_reversed_border(&self) -> bool {
+        self.0.reversed_border
+    }
+
+    #[setter]
+    pub fn set_reversed_border(&mut self, value: bool) {
+        self.0.reversed_border = value;
+    }
+
+    #[getter]
+    pub fn get_min_tag_width(&self) -> usize {
+        self.0.min_tag_width
+    }
+
+    #[setter]
+    pub fn set_min_tag_width(&mut self, value: usize) {
+        self.0.min_tag_width = value;
+    }
+
+    #[getter]
+    pub fn get_min_white_black_difference(&self) -> u8 {
+        self.0.min_white_black_difference
+    }
+
+    #[setter]
+    pub fn set_min_white_black_difference(&mut self, value: u8) {
+        self.0.min_white_black_difference = value;
+    }
+
+    #[getter]
+    pub fn get_downscale_factor(&self) -> usize {
+        self.0.downscale_factor
+    }
+
+    #[setter]
+    pub fn set_downscale_factor(&mut self, value: usize) {
+        self.0.downscale_factor = value;
+    }
+
+    #[staticmethod]
+    pub fn all() -> Self {
+        Self(DecodeTagsConfig::all())
+    }
+}
+
+#[pyclass(name = "FitQuadConfig", eq, get_all, set_all)]
+#[derive(Default, PartialEq, Clone)]
+pub struct PyFitQuadConfig {
+    pub cos_critical_rad: f32,
+    pub max_line_fit_mse: f32,
+    pub max_nmaxima: usize,
+    pub min_cluster_pixels: usize,
+}
+
+#[pymethods]
+impl PyFitQuadConfig {
+    #[new]
+    pub fn new(
+        cos_critical_rad: f32,
+        max_line_fit_mse: f32,
+        max_nmaxima: usize,
+        min_cluster_pixels: usize,
+    ) -> Self {
+        Self {
+            cos_critical_rad,
+            max_line_fit_mse,
+            max_nmaxima,
+            min_cluster_pixels,
+        }
+    }
+}
+
+#[pyclass(name = "AprilTagDecoder")]
+pub struct PyAprilTagDecoder(AprilTagDecoder);
+
+#[pymethods]
+impl PyAprilTagDecoder {
+    #[new]
+    pub fn new(config: Py<PyDecodeTagsConfig>, img_size: PyImageSize) -> PyResult<Self> {
+        Python::attach(|py| {
+            let config = config.borrow(py);
+
+            Ok(Self(
+                AprilTagDecoder::new(config.0.clone(), img_size.into())
+                    .map_err(|err| PyErr::new::<PyException, _>(err.to_string()))?,
+            ))
+        })
+    }
+
+    pub fn clear(&mut self) {
+        self.0.clear();
+    }
+
+    pub fn decode(&mut self, src: PyImage) -> PyResult<Vec<PyDetection>> {
+        let img: Image<u8, 1, _> = Image::from_pyimage(src)
+            .map_err(|err| PyErr::new::<PyException, _>(err.to_string()))?;
+
+        let detection = self
+            .0
+            .decode(&img)
+            .map_err(|err| PyErr::new::<PyException, _>(err.to_string()))?;
+
+        let py_detection = detection.iter().map(|d| d.clone().into()).collect();
+        Ok(py_detection)
+    }
+}
+
+#[pyclass(name = "Detection", eq, get_all, set_all)]
+#[derive(PartialEq, Clone)]
+pub struct PyDetection {
```

Comment:
I think we should name this as ApriltagDetection to avoid confusion with other types, possibly upstream too